### PR TITLE
split out and expose package recipe

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
this allows building underlying derivations with other inputs:

    nix-repl> :l https://github.com/nixos/nixpkgs/tarball/nixpkgs-unstable
    nix-repl> :l https://github.com/nix-community/nixdoc/tarball/master
    nix-repl> pkgs.callPackage recipes.nixdoc {}

### CODE REVIEW

File: package.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{ lib, rustPlatform }:
```
Comment (infinisil) [inline package.nix:1]: Would you want this file to be a stable interface? If so it should have a comment here (and if not, I don't see a reason for doing this as is)

Comment (fricklerhandwerk) [reply package.nix:1]: Hm no, I'd rather have the recipe as an exported attribute in `default.nix`, but that would require tearing apart the whole setup.

Comment (fricklerhandwerk) [reply package.nix]: Gave it another thought.

File: flake.nix [hunk 2]
```diff
@@ -10,24 +10,32 @@
...
+        rustPlatform.buildRustPackage {
+          pname = package.name;
+          version = package.version;
+          src = ./.;
+          cargoLock.lockFile = ./Cargo.lock;
+        };
+    in
+    {
+      recipes.default = recipe;
+      recipes.nixdoc = recipe;
```
Comment (infinisil) [inline flake.nix:27]:
```suggestion
```

Doesn't seem necessary

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by infinisil** [comment]
This seems fine to me, even if not very standard. If you could add a small changelog entry for this that would be great, since this is essentially a new part of the API.

**Review by infinisil — APPROVED** [review]

**Pull request merged**